### PR TITLE
Updating docs on how to extend the tokens

### PR DESCRIPTION
**Pull request by mltejera**
Leave's Shift's existing migration docs in place so as to not break any current links.

Copies the whole document to the current docs folder.

Takes a subset of that out and moves it to 
- The patterns folder
- The Storybook theming section (these new docs are The Way)

### CODE REVIEW

**Review by khmakoto — APPROVED** [review]

File: docs/react-v9/contributing/migration-notes.md [hunk 1]
```diff
@@ -0,0 +1,407 @@
+# Migration notes from `@fluentui/react-components@9.0.0-beta.5`to `@fluentui/react-components@9.0.0-rc.1`
```
Comment (layershifter) [inline docs/react-v9/contributing/migration-notes.md:1]: Not sure that we want to this document in a new location. It was good in its time, some info in it is already outdated.

File: apps/public-docsite-v9/src/Concepts/Theming.stories.mdx [hunk 2]
```diff
@@ -123,25 +123,54 @@ export const customLightTheme: Theme = {
 
 ### Extending theme with new tokens
 
-Similarly to overriding existing tokens, you can add custom tokens as well.
+It's often useful for an app to extend the base set of tokens from Fluent UI.
 
-⚠ Components which use custom tokens cannot be shared between applications. Keep in mind that any application which uses a component with custom tokens must also add the custom tokens to its own themes. Instead of adding custom tokens inside potentially reusable components, you should talk to design.
+⚠ Components in this repo should _not_ do this.
```
Comment (layershifter) [inline apps/public-docsite-v9/src/Concepts/Theming.stories.mdx:128]: It's in public docsite, so we probably don't need this

File: apps/public-docsite-v9/src/Concepts/Theming.stories.mdx [hunk 3]
```diff
@@ -123,25 +123,54 @@ export const customLightTheme: Theme = {
...
 
-⚠ Keep in mind that the object generated by the `themeToTokensObject()` will contain all the tokens and will not be tree-shakeable.
+```
 
-```tsx
-import { themeToTokensObject } from '@fluentui/react-components';
+```
 
-export const customTokens = themeToTokensObject(customLightTheme);
 ```
```
Comment (layershifter) [inline apps/public-docsite-v9/src/Concepts/Theming.stories.mdx:176]: Delete this empty blocks?

**Review by layershifter — APPROVED** [review]

**Review by mainframev — CHANGES_REQUESTED** [review]
A few typos only and suggestion is up to you 🙂

File: docs/react-v9/contributing/migration-notes.md [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+-  focusOutline2: createFocusOutlineStyle(theme => ({ backgroundColor: theme.colorNeutralBackground1 })),
++  focusOutline2: createFocusOutlineStyle({ backgroundColor: tokens.colorNeutralBackground1 }),
+});
+```
+
+For more details, please check [microsoft/fluentui#20651](https://github.com/microsoft/fluentui/pull/20651).
+
+### CSS shorthands no longer supported
+
+[CSS shorthands](https://developer.mozilla.org/en-US/docs/Web/CSS/Shorthand_properties) in `makeStyles()` calls are no longer supported. For many shorthands there exist matching functions in `@fluentui/react-components`:
```
Comment (spmonahan) [inline docs/react-v9/contributing/migration-notes.md:40]: This isn't really accurate anymore right? https://griffel.js.org/react/guides/limitations#some-of-css-shorthands-are-not-supported

I think we should update this.

File: docs/react-v9/contributing/migration-notes.md (hunk 1 above)
Comment (spmonahan) [reply docs/react-v9/contributing/migration-notes.md:1]: Yeah agreed. The shorthands info is definitely out of date.

Comment (mltejera) [reply docs/react-v9/contributing/migration-notes.md]: Removed this doc from the PR.

File: apps/public-docsite-v9/src/Concepts/Theming.stories.mdx (hunk 3 above)
Comment (mltejera) [reply apps/public-docsite-v9/src/Concepts/Theming.stories.mdx:176]: Good catch - removed

**Review by mainframev — APPROVED** [review]

**Pull request merged**